govuk_chat_evaluation/rag_answers/handle_model_id_collisions.py

```diff
@@ -0,0 +1,19 @@
+from collections import defaultdict
+from typing import TypeVar, List
+from .data_models import GenerateInput
+
+InputType = TypeVar("InputType", bound=GenerateInput)
+
+
+def ensure_unique_model_ids(inputs: List[InputType]) -> List[InputType]:
```

Comment:
I don't think you need this to return a value anymore.

I think as EvaluationTestCase is a subclass of GenerateInput you can probably just have a type of list[GenerateInput], rather than needing to use a generic, but could be that you need to do Sequence[GenerateInput] I have a recollection of there being sensitivity.